### DISCUSSION TITLE
Ractor + Tower

### DISCUSSION THREAD

**Discussion by EnvOut** [slawlor/ractor#192] [Q&A]
I'm currently exploring options for integrating Ractor with Tower. I'm particularly interested in moving requests to a specific actor. 
Could you please provide information or guidance on how this integration can be achieved?

**Comment by slawlor** [comment]
> I'm currently exploring options for integrating Ractor with Tower. I'm particularly interested in moving requests to a specific actor. 
> Could you please provide information or guidance on how this integration can be achieved? 

What do you expect to get out of this kind of integration? Do you have a design in mind? Ractor could be the underlying request runtime but if you want it as a middleware then I would suggest probably wrapping a factory.

Something like 

```rust
struct MyMiddleware {
factory: ActorRef<Message,...>
}
```

And you would implement the tower traits I guess which then send rpcs to the factor workers. Something like that would make sense, and would let you implement back pressure on your middleware.

That is assuming I understand how Tower works, I don't personally have any experience with it yet

**Reply by rajivr** [reply]
> That is assuming I understand how Tower works, I don't personally have any experience with it yet

Hi @slawlor , Thanks for creating ractor. I just finished exploring ractor APIs.

I was wondering if you had a chance to explore Tower/Axum since the last update to this thread?

If we look at [this](https://github.com/tokio-rs/axum/blob/axum-v0.8.2/examples/serve-with-hyper/src/main.rs) example from Axum, we can see that two Tokio tasks are spawned (second one happens in `server::conn::auto::Builder::new(TokioExecutor::new())`) every time a connection is accepted.

Could you please suggest what would be the idiomatic Ractor approach here? Instead of `loop`, should we have an actor that only accepts the connection, and then dispatches to a factory?

**Reply by slawlor** [reply]
I haven't and not using it personally, I don't really have motivation or the time to. I don't even have a test environment using anything with Tower or Axum. But you're welcome to use my suggestions I posted above and try and do it yourself, or try and get the community to get involved. But I doubt I'll have time in the foreseeable future to do it myself
